### PR TITLE
color grouping: move regexfilter attribute from runUi to runData

### PR DESCRIPTION
**Pull request by japie1235813**
* Motivation for features / changes
One of the two color grouping by regex ui option is to align the filter run regex and color grouping regex.
Move the regexFilter attribute from ui state to data state.

### CODE REVIEW

File: tensorboard/webapp/runs/store/runs_types.ts [hunk 1]
```diff
@@ -52,6 +52,8 @@ export interface RunsDataRoutefulState {
   groupKeyToColorString: Map<string, string>;
   userSetGroupBy?: GroupBy;
```
Comment (stephanwlee) [inline tensorboard/webapp/runs/store/runs_types.ts:53]: You may want to change this to `GroupByKey` so there is no redundant information in the `colorGroupRegexString`.

**Review by stephanwlee — COMMENTED** [review]
Looking great!

File: tensorboard/webapp/runs/store/runs_types.ts (hunk 1 above)
Comment (japie1235813) [reply tensorboard/webapp/runs/store/runs_types.ts:53]: Agree. Yet this changes includes `groupRuns` method changes and a bunch of utils(group by regex) and tests changes. Would like to separate them into another pr.

Comment (stephanwlee) [reply tensorboard/webapp/runs/store/runs_types.ts:53]: Another PR sounds good but do note that we don't want to change API of utils, actions, and selectors.

**Review by stephanwlee — APPROVED** [review]

**Pull request merged**